src/components/CustomerQuota/NoQuota/NoQuotaCard.tsx

```diff
@@ -0,0 +1,210 @@
+ * Precondition: Only rendered when all items in cart have max quantity of 0
+ */
+export const NoQuotaCard: FunctionComponent<NoQuotaCard> = ({
+  ids,
+  cart,
+  onCancel,
+  onAppeal
+}) => {
+  const [isShowFullList, setIsShowFullList] = useState<boolean>(false);
+  const { getProduct, allProducts } = useProductContext();
+  const { token, endpoint } = useAuthenticationContext();
+
+  const policyType = cart.length > 0 && getProduct(cart[0].category)?.type;
+
+  const { pastTransactionsResult } = usePastTransaction(ids, token, endpoint);
+  // Assumes results are already sorted (valid assumption for results from /transactions/history)
+  const sortedTransactions = pastTransactionsResult?.pastTransactions;
+
+  const latestTransactionTime: Date | undefined =
+    sortedTransactions?.[0].transactionTime ?? undefined;
+  const now = new Date();
+  const secondsFromLatestTransaction = latestTransactionTime
+    ? differenceInSeconds(now, latestTransactionTime)
+    : -1;
+
+  const hasAppealProduct = (): boolean => {
+    return allProducts.some(policy => policy.categoryType === "APPEAL");
+  };
+
+  // Group transactions by category
+  const transactionsByCategoryMap: {
+    [category: string]: {
+      transactions: Transaction[];
+      hasLatestTransaction?: boolean;
+    };
+  } = {};
+  sortedTransactions?.forEach(item => {
+    const policy = allProducts.find(
+      policy => policy.category === item.category
+    );
+    const categoryName = policy?.name ?? item.category;
+    const formattedDate = format(item.transactionTime, "d MMM yyyy, h:mma");
+
+    if (!(categoryName in transactionsByCategoryMap)) {
+      transactionsByCategoryMap[categoryName] = {
+        transactions: [],
+        hasLatestTransaction: false
+      };
+    }
+    transactionsByCategoryMap[categoryName].transactions.push({
+      transactionDate: formattedDate,
+      details: getAllIdentifierInputDisplay(item.identifierInputs ?? []),
+      quantity: formatQuantityText(
+        item.quantity,
+        policy?.quantity.unit || { type: "POSTFIX", label: " qty" }
+      ),
+      isAppeal: policy?.categoryType === "APPEAL"
+    });
+    transactionsByCategoryMap[categoryName].hasLatestTransaction =
+      transactionsByCategoryMap[categoryName].hasLatestTransaction ||
+      differenceInSeconds(latestTransactionTime || 0, item.transactionTime) ===
+        0;
+  });
+
+  // Split categories into those with the latest transactions and those without
+  const latestTransactionsByCategory: TransactionsByCategory[] = [];
+  const otherTransactionsByCategory: TransactionsByCategory[] = [];
+  Object.entries(transactionsByCategoryMap).forEach(([key, value]) => {
+    const { transactions, hasLatestTransaction } = value;
+    if (hasLatestTransaction) {
+      latestTransactionsByCategory.push({
+        category: key,
+        transactions: transactions
+      });
+    } else {
+      otherTransactionsByCategory.push({ category: key, transactions });
+    }
+  });
```

Comment:
Hmmm I'm not very clear what this part does together with the concatenation below. Is there are better way to phrase the purpose of these 2 blocks? HMW test these too?

Reply:
Does my explanation about the requirements clear things up for you?

>For category, Eida said to order like that:
>
>Category with latest transaction
>All other categories by alphabetical order

Same suggestion about moving to util function but keep in same file?

Reply:
Yea would be best if we can add some tests here (:

Reply:
Done!! Broke them up in a way that I think makes sense, do let me know if you think can be broken down more :)